### PR TITLE
fix: Container with empty content has incorrect header border radius

### PR DESCRIPTION
**Pull request by YueyingLu**
### Description

When container has no content and disableContentPaddings, the bottom border radius are overlapped by header. The change adds border radius when no content in the container, as we did for collapsed ExpandableSection. 
For test, add one more instance in simple.page, it will show in the screenshot tests. 

Related links, issue AWSUI-20171

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/container/index.tsx [hunk 1]
```diff
@@ -24,6 +24,7 @@ export default function Container({
       disableContentPaddings={disableContentPaddings}
       {...externalProps}
       {...baseComponentProps}
+      __hiddenContent={disableContentPaddings && !props.children}
```
Comment (abdhalees) [inline src/container/index.tsx:27]: is `disableContentPaddings` necessary? 
this should work regardless of padding, customers used this as a hack but I think it should work without it

Comment (YueyingLu) [reply src/container/index.tsx:27]: Makes sense! Will update it.

File: pages/container/simple.page.tsx [hunk 2]
```diff
@@ -102,6 +102,10 @@ export default function SimpleContainers() {
               Content area takes the available vertical space
             </Container>
           </div>
+          <Container
+            disableContentPaddings={true}
```
[the remark this answers is not in the document]
Comment (YueyingLu) [reply pages/container/simple.page.tsx:106]: I would rather leave it here because without disableContentPaddings the border looks correct. It is to make sure the fix still works even with disableContentPaddings.

**Review by abdhalees — APPROVED** [review]

File: src/container/index.tsx [hunk 3]
```diff
@@ -24,6 +24,7 @@ export default function Container({
       disableContentPaddings={disableContentPaddings}
       {...externalProps}
       {...baseComponentProps}
+      __hiddenContent={!props.children}
```
Comment (just-boris) [inline src/container/index.tsx:27]: Does this mean that all internal use of container remain unfixed? Why is that needed?

This fix should be applied to internal instances too

Comment (just-boris) [reply src/container/index.tsx:27]: Basically, [this line](https://github.com/cloudscape-design/components/blob/main/src/container/internal.tsx#L104) should become

```
[styles['with-hidden-content']]: !children || __hiddenContent
```

Comment (YueyingLu) [reply src/container/index.tsx:27]: Hi Boris, thanks for checking it. I was not sure we should not use container without content internally at all, It should be either ExpandableSection or Header? But I agree it makes sense the fix should be for all cases.

Comment (YueyingLu) [reply src/container/index.tsx:27]: fix in https://github.com/cloudscape-design/components/pull/669
